File: backend/schemas/dividends.py
from pydantic import BaseModel


class DividendItem(BaseModel):
    etf_ticker: str
    ex_date: str
    dividend_amount: float
    currency: str


class DividendsResponse(BaseModel):
    dividends: list[DividendItem]
    total: int
    page: int
    page_size: int
    total_pages: int

File: frontend/src/types/dividends.ts
export interface DividendItem {
  etf_ticker: string
  ex_date: string
  dividend_amount: number
  currency: string
}

export interface DividendsResponse {
  dividends: DividendItem[]
  total: number
  page: number
  page_size: number
  total_pages: number
}
